lib/querystring.js

```diff
@@ -277,20 +277,23 @@ QueryString.parse = QueryString.decode = function(qs, sep, eq, options) {
           key = decodeStr(key, decode);
         if (valEncoded)
           value = decodeStr(value, decode);
-        // Use a key array lookup instead of using hasOwnProperty(), which is
-        // slower
-        if (keys.indexOf(key) === -1) {
-          obj[key] = value;
-          keys[keys.length] = key;
-        } else {
-          const curValue = obj[key];
-          // `instanceof Array` is used instead of Array.isArray() because it
-          // is ~15-20% faster with v8 4.7 and is safe to use because we are
-          // using it with values being created within this function
-          if (curValue instanceof Array)
-            curValue[curValue.length] = value;
-          else
-            obj[key] = [curValue, value];
+
+        if (key.length > 0 || value.length > 0) {
```

Comment:
Just took a look at the spec and this bit of code, I think the current way is the best :D. The only one that needs to be ignored is `&&`, otherwise it doesn't match what the spec needs.